[ML] Add per-partition categorization option (#57683)
elastic/elasticsearch: docs/reference/ml/anomaly-detection/apis/get-category.asciidoc

`<category_id>`::

> (Optional, long) Identifier for the category. If you do not specify this

Reviewer: > If you do not specify this parameter...

I found this sentence a bit long, so I'd probably split it. For example, something like this:

If you specify neither the category ID nor the `partition_field_value`, the API returns information about all categories. If you specify only the `partition_field_value`, the API returns information about all categories for the specified partition.

parameter, the API returns information about all categories in the {anomaly-job},
or all categories in the {anomaly-job} for the specified partition if

> `partition_field_value` is supplied. Since `category_id` is unique at the job

Reviewer: > Since `category_id` is unique at the job...

IMO, this sentence is not necessary if you just change first phrase like this:

> Identifier for the category, which is unique in the job.

The value is increased by 10% to enable matching for similar fields that have
not been analyzed.

// This doesn't use the shared description because there are
// categorization-specific aspects to its use in this context
`partition_field_name`::

> (string) The field used to segment the categorization, if per-partition

Reviewer: We generally put the conditional bit at the beginning of the sentence, so folks can stop reading if it doesn't apply.  I'd suggest:

> If per-partition categorization is enabled, this value identifies the field used to segment the categorization.

> categorization was enabled. This field is not present when per-partition

Reviewer: To avoid confusion with the "field" in the previous sentence, I'd recommend omitting the second sentence or changing it like this:
```suggestion
categorization was enabled. It is not present when per-partition
```
